### PR TITLE
Raise minimum supported Swift version from 5.4 to 5.5

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

SwiftNIO periodically drops support for older Swift versions. Now that 5.7 has been released, 5.4 will be dropped.

Modifications:

- Remove 5.4 specific Package.swift and docker-compose
- Update the 5.7 docker-compose to use the released 5.7 and move from focal (2004) to jammy (2204)
- Remove unused swiftformat from Dockerfile
- Update tools version in syscall wrapper tests to 5.5
- Update docs

Results:

Minimum Swift version is 5.5

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -4,19 +4,21 @@ This document specifies the security process for the SwiftNIO project.
...
+If a hypothetical security vulnerability was introduced in 2.10.0, then SwiftNIO core
 team would create the following patch releases:
 
 * NIO 2.29. + plus next patch release to address the issue for projects that support
   Swift 5.0 and 5.1
-* NIO 2.39. + plus next patch release to address the issue for projects that support 
+* NIO 2.39. + plus next patch release to address the issue for projects that support
   Swift 5.2 and 5.3
-* mainline + plus next patch release to address the issue for projects that support 
+* NIO 2.41. + plus next patch release to address the issue for projects that support
```
Comment (Lukasa) [inline SECURITY.md:18]: 2.42, I think.

**Review by Lukasa — CHANGES_REQUESTED** [review]

**Comment by glbrntt** [comment]
This change appears to have cause allocations to change in 5.5 and 5.6 (and presumably 5.7).  `1_reqs_1000_conn` has regressed 👀...

**Comment by glbrntt** [comment]
> This change appears to have cause allocations to change in 5.5 and 5.6 (and presumably 5.7). `1_reqs_1000_conn` has regressed 👀...

These must come from [GHSA-7fj7-39wj-c64f](https://github.com/apple/swift-nio/commit/a16e2f54a25b2af217044e5168997009a505930f). Fixing in a separate PR.

**Comment by yim-lee** [comment]
CI (including performance jobs) updated. @swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Pull request merged**